lib/internal/http2/core.js

```diff
@@ -1077,6 +1077,8 @@ function onHandleFinish() {
   if (this[kID] === undefined) {
     this.once('ready', onHandleFinish.bind(this));
   } else {
+    if (this[kState].headersSent === false)
+      this.rstWithNoError();
```

Comment:
What I would expect in that situation is a proper http 200 response with no data. An rst is a bit of a special case that should only be sent when explicitly necessary